# Construction of pYPK0_TPI1tp_CiGXF1_ENO2tp (pMEC1134)

[pYPKa_Z_TPI1tp](pYPKa_Z_TPI1tp.ipynb)

[pYPKa_E_ENO2tp](pYPKa_E_ENO2tp.ipynb)

In [1]:
import pydna
from pydna import ipynb_importer

In [2]:
from ypk_std_primers import p166, p167, p567, p568

[Yeast Pathway Kit Standard Primers](ypk_std_primers.ipynb)

In [3]:
from Bio.Restriction import ZraI, AjiI, EcoRV

In [4]:
pYPK0 = pydna.read("pYPK0.gb")

In [5]:
promoter_clone = pYPKa_Z_TPI1tp = pydna.read("pYPKa_Z_TPI1tp.gb")

[Genbank](http://www.ncbi.nlm.nih.gov/nucleotide/AJ937350.1) record

In [6]:
from IPython.display import HTML
HTML('<iframe src=http://www.ncbi.nlm.nih.gov/pubmed/16402921 width=100% height=250></iframe>')

In [7]:
gb = pydna.Genbank("bjornjobbb@gmail.com")
gene_template = gb.nucleotide("AJ937350.1")

In [8]:
gene_template.list_features()

+----------+-----------+-------+------+--------+--------------+--------+------+
| Feature# | Direction | Start | End  | Length | id           | type   | orf? |
+----------+-----------+-------+------+--------+--------------+--------+------+
| 0        |    -->    |   0   | 1644 |   1644 | <unknown id> | source | yes  |
| 1        |    -->    |   0   | 1644 |   1644 | <unknown id> | gene   | yes  |
| 2        |    -->    |   0   | 1644 |   1644 | <unknown id> | CDS    | yes  |
+----------+-----------+-------+------+--------+--------------+--------+------+

In [9]:
orf = gene_template.extract_feature(2)

In [10]:
orf.isorf()

True

In [11]:
print len(orf)
orf.seguid()

1644


nRdsznd4WZFAGP2u0IIooDFQS90

In [12]:
print str(orf.seq.translate())

MSQDSHSSGAATPVNGSILEKEKEDSPVLQVDAPQKGFKDYIVISIFCFMVAFGGFVFGFDTGTISGFVNMSDFKDRFGQHHADGTPYLSDVRVGLMISIFNVGCAVGGIFLCKVADVWGRRIGLMFSMAVYVVGIIIQISSSTKWYQFFIGRLIAGLAVGTVSVVSPLFISEVSPKQIRGTLVCCFQLCITLGIFLGYCTTYGTKTYTDSRQWRIPLGLCFAWAILLVVGMLNMPESPRYLVEKHRIDEAKRSIARSNKIPEEDPFVYTEVQLIQAGIEREALAGQASWKELITGKPKIFRRVIMGIMLQSLQQLTGDNYFFYYGTTIFQAVGLKDSFQTSIILGIVNFASTFVGIYVIERLGRRLCLLTGSAAMFICFIIYSLIGTQHLYKQGYSNETSNTYKASGNAMIFITCLYIFFFASTWAGGVYCIISESYPLRIRSKAMSIATAANWLWGFLISFFTPFITSAIHFYYGFVFTGCLAFSFFYVYFFVYETKGLSLEEVDEMYASGVLPLKSASWVPPNLEHMAHSAGYAGADKATDEQV*


In [13]:
pf, pr = pydna.parse('''>580_GXF1_YPK_fwd (83-mer)
                        gtcgaggaacgccaggttgcccactttctcactagtgaAAatgtcacaagattcgcattctt

                        >579_GXF1_YPK_rev (83-mer)
                        ATTTAAatcctgatgcgtttgtctgcacagatggcgcgttaaacctgttcgtcggtggcctt''')

In [14]:
g = pydna.pcr( pf, pr, orf)

In [15]:
g.figure()

                                        5ATGTCACAAGATTCGCATTCTT...AAGGCCACCGACGAACAGGTTTAA3
                                                                  |||||||||||||||||||||||| tm 58.8 (dbd) 72.9
                                                                 3ttccggtggctgcttgtccaaattgcgcggtagacacgtctgtttgcgtagtcctaAATTTA5
5gtcgaggaacgccaggttgcccactttctcactagtgaAAatgtcacaagattcgcattctt3
                                         |||||||||||||||||||||| tm 50.9 (dbd) 63.9
                                        3TACAGTGTTCTAAGCGTAAGAA...TTCCGGTGGCTGCTTGTCCAAATT5

In [17]:
terminator_clone = pYPKa_E_ENO2tp = pydna.read("pYPKa_E_ENO2tp.gb")

In [18]:
p  = pydna.pcr( p167, p567, promoter_clone)
t  = pydna.pcr( p568, p166, terminator_clone)

In [19]:
pYPK0_E_Z, stuffer = pYPK0.cut((EcoRV, ZraI))

In [20]:
(pYPK0_E_Z, p, g, t)

(Dseqrecord(-5681), Amplicon(744), Amplicon(1722), Amplicon(625))

In [21]:
asm = pydna.Assembly((pYPK0_E_Z, p, g, t), limit=31)

In [22]:
asm

Assembly:
Sequences........................: [5681] [744] [1722] [625]
Sequences with shared homologies.: [5681] [744] [625] [1722]
Homology limit (bp)..............: 31
Number of overlaps...............: 4
Nodes in graph(incl. 5' & 3')....: 6
Only terminal overlaps...........: No
Circular products................: [8526]
Linear products..................: [8624] [8587] [8576] [8563] [7999] [7930] [6891] [6327] [6245] [3004] [2416] [2310] [98] [61] [50] [37]

In [23]:
candidate = asm.circular_products[0]
candidate.figure()

 -|pYKP0|98
|        \/
|        /\
|        98|744bp_PCR_prod|38
|                          \/
|                          /\
|                          38|1722bp_PCR_prod|35
|                                             \/
|                                             /\
|                                             35|625bp_PCR_prod|61
|                                                               \/
|                                                               /\
|                                                               61-
|                                                                  |
 ------------------------------------------------------------------

In [38]:
result = candidate.synced(pYPK0)

The new construct should have cseguid ``` 0t3SB9dac_rF8FmwUiA95T8KlRA ``` and 8526 bp.

In [39]:
print len(result)
result.cseguid()

8526


0t3SB9dac_rF8FmwUiA95T8KlRA

In [40]:
result.name="pMEC1134"
result.description = "pYPK0_TPI1tp_CiGXF1_ENO2tp"
result.stamp()

cSEGUID_0t3SB9dac_rF8FmwUiA95T8KlRA_2015-05-10T09:30:50.558456

In [41]:
result.write("pYPK0_TPI1tp_CiGXF1_ENO2tp.gb")

###[Download](pYPK0_TPI1tp_CiGXF1_ENO2tp.gb) genbank file.

In [42]:
reloaded = pydna.read("pYPK0_TPI1tp_CiGXF1_ENO2tp.gb")

In [43]:
reloaded.cseguid()

0t3SB9dac_rF8FmwUiA95T8KlRA

In [44]:
reloaded.description

'<unknown description> cSEGUID_0t3SB9dac_rF8FmwUiA95T8KlRA_2015-05-10T09:27:17.842828'

[old](pYPK0_TPI1tp_CiGXF1_ENO2tp_OLD.gb)